### ISSUE TITLE
IPY340 MWrite.pushbutton pickle module TypeError

### ISSUE THREAD

**Issue by nodatasheet** [eirannejad/pyRevit#1704]
**Describe the bug**
With IronPython 3.4 engine pressing on [MWrite.pushbutton](https://github.com/eirannejad/pyRevit/blob/develop/extensions/pyRevitTools.extension/pyRevit.tab/Selection.panel/mema.stack/MWrite.pushbutton/script.py), [MRead.pushbutton](https://github.com/eirannejad/pyRevit/blob/develop/extensions/pyRevitTools.extension/pyRevit.tab/Selection.panel/mema.stack/MRead.pushbutton/script.py) or [MAppend.pushbutton](https://github.com/eirannejad/pyRevit/blob/develop/extensions/pyRevitTools.extension/pyRevit.tab/Selection.panel/mema.stack/MAppend.pushbutton/script.py) rises `TypeError: must be unicode, not bytes`

<details>
<summary><b>Error traceback</b></summary>

```python
IronPython Traceback:
Traceback (most recent call last):
 File "C:\Users\Konstantin\AppData\Roaming\pyRevit\pyRevit\develop\extensions\pyRevitTools.extension\pyRevit.tab\Selection.panel\mema.stack\MWrite.pushbutton\script.py", line 13, in <module>
 File "pickle.py", line 1538, in _dump
 File "pickle.py", line 409, in dump
 File "pickle.py", line 216, in write
TypeError: must be unicode, not bytes
<...>
```
</details>

**To Reproduce**
1. Select elements in Revit
2. Click on `MWrite` push button
3. See error

**Expected behavior**
Add current selection to memory

**Screenshots**
no

**Desktop (please complete the following information):**
 - OS: Windows 11
 - pyRevit Version 4.8.13:16d0938
 - pyRevit Environment: 
    <details>
    
    <summary>pyrevit env</summary>
    
    
    ```cmd
    ==> Registered Clones (full git repos)
    develop | Branch: "develop" | Version: "4.8.13:16d0938" | Path: "C:\Users\Konstantin\AppData\Roaming\pyRevit\pyRevit\develop"
    master | Branch: "master" | Version: "4.8.12.22247+0031:b7704ea" | Path: "C:\Users\Konstantin\AppData\Roaming\pyRevit\pyRevit\master"
    ==> Registered Clones (deployed from archive/image)
    ==> Attachments
    develop | Product: "Autodesk Revit 2023" | Engine: IPY340PR (340) | Path: "C:\Users\Konstantin\AppData\Roaming\pyRevit\pyRevit\develop"
    develop | Product: "Autodesk Revit 2022" | Engine: IPY340PR (340) | Path: "C:\Users\Konstantin\AppData\Roaming\pyRevit\pyRevit\develop"
    ==> Installed Extensions
    pyKostik | Type: Unknown | Repo: "" | Installed: "...\pyKostik\pyKostik.extension"
    ==> Default Extension Search Path
    C:\Users\Konstantin\AppData\Roaming\pyRevit\Extensions
    ==> Extension Search Paths
    ...\pyKostik
    ==> Extension Sources - Default
    https://github.com/eirannejad/pyRevit/raw/master/extensions/extensions.json
    ==> Extension Sources - Additional
    ==> Installed Revits
    Autodesk Revit 2023 | Version: 23.1.10.4 | Build: 20221122_1550(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2023\"
    Autodesk Revit 2022 | Version: 22.1.30.34 | Build: 20220520_1515(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2022\"
    ==> Running Revit Instances
    Error: Object reference not set to an instance of an object.
    Run with "--debug" option to see debug messages
    ```
    
    </details>

**Additional context**
I assume the reason is that in Python 3, [binary modes](https://docs.python.org/3.8/glossary.html#term-binary-file) `'wb'` and `'rb'` must be specified during the `open` operation.

Changing `'w'`, `'r'` to `'wb'`, `'rb'` in [MWrite.pushbutton](https://github.com/eirannejad/pyRevit/blob/16d093826ab7f6448165daf8ef09b13f959c3fb8/extensions/pyRevitTools.extension/pyRevit.tab/Selection.panel/mema.stack/MWrite.pushbutton/script.py#L12), [MRead.pushbutton](https://github.com/eirannejad/pyRevit/blob/16d093826ab7f6448165daf8ef09b13f959c3fb8/extensions/pyRevitTools.extension/pyRevit.tab/Selection.panel/mema.stack/MRead.pushbutton/script.py#L14) and [MAppend.pushbutton](https://github.com/eirannejad/pyRevit/blob/16d093826ab7f6448165daf8ef09b13f959c3fb8/extensions/pyRevitTools.extension/pyRevit.tab/Selection.panel/mema.stack/MAppend.pushbutton/script.py#L14-L21) solved the issue for me.

**Comment by thumDer** [comment]
How about making a PR for this one too? It would be appreciated ❤️
And thanks for checking these!

**Comment by jmcouffin** [comment]
Fixed by #1765 
Thanks @nodatasheet
